engine/src/main/java/com/datatorrent/stram/StramClient.java

```diff
@@ -287,7 +287,15 @@ public void copyInitialState(Path origAppDir) throws IOException
     FileStatus[] lFiles = fs.listStatus(origAppDir);
     for (FileStatus f : lFiles) {
       if (f.isDirectory()) {
-        String targetPath = f.getPath().toString().replace(origAppDir.toString(), newAppDir);
+
+        // In case of MapR/MapR-FS, f.getPath().toString() returns path as maprfs:///<orig app dir>
+        // whereas origAppDir.toString & newAppDir are in maprfs:/<orig or new app dir> format
+        // e.g.
+        // f.getPath().toString -> maprfs:///user/dtadmin/datatorrent/apps/application_1481890072066_0004/checkpoints
+        // origAppDir -> maprfs:/user/dtadmin/datatorrent/apps/application_1481890072066_0004
+        // newAppDir -> maprfs:/user/dtadmin/datatorrent/apps/application_1481890072066_0005
+
+        String targetPath = f.getPath().toString().replace("///","/").replace(origAppDir.toString(), newAppDir);
```

Comment:
Instead of trying to do this workaround, I suggest we try replacing only the "path" part of the path. You could do it the following way

String origAppPath = getPath(origAppDir);
String newAppPath = getPath(new Path(newAppDir));
String targetPath = f.getPath().toString().replace(origAppPath, newAppPath);

private String getPath(Path path) {
    Path npath = Path.getPathWithoutSchemeAndAuthority(path);
    return npath.toString();
}